tests/test_worker.py

```diff
@@ -122,6 +134,47 @@ async def test_sleeper(self):
         self.assertEqual(job.status, Status.FAILED)
         assert "TimeoutError" in job.error
 
+    async def test_heartbeat(self):
+        job = await self.queue.enqueue("monitored_sleeper", heartbeat=2, timeout=None)
+        self.worker.context["sleep"] = 4
+        task = asyncio.create_task(self.worker.process())
+        # wait for the job to actually start
+        def callback(job_key, status):
+            self.assertEqual(job.key, job_key)
+            self.assertEqual(status, Status.ACTIVE)
+            return True
+
+        await self.queue.listen([job.key], callback)
+        await asyncio.sleep(2)
+        swept = await self.queue.sweep()
+        self.assertNotIn(
+            set(swept),
+            {job.id.encode()},
+        )
+        await task
+        await job.refresh()
+        self.assertEqual(job.status, Status.COMPLETE)
+
+    async def test_missed_heartbeat(self):
```

Comment:
There's also this test: https://github.com/tobymao/saq/blob/master/tests/test_queue.py#L235

That test is a bit more "white box" compared to this one.

But do we need both?